rocketpy/control/controller.py

```diff
@@ -88,7 +94,44 @@ def __init__(
         else:
             self.observed_variables = []
 
-    def __call__(self, time, state_vector, state_history):
+    def __init_controller_function(self, controller_function):
+        """Checks number of arguments of the controller function and initializes
+        it with the correct number of arguments. This is a workaround to allow
+        the controller function to receive sensors without breaking changes"""
+        sig = signature(controller_function)
+        if len(sig.parameters) == 6:
+
+            def new_controller_function(
+                time,
+                sampling_rate,
+                state_vector,
+                state_history,
+                observed_variables,
+                interactive_objects,
+                sensors,
+            ):
+                return controller_function(
+                    time,
+                    sampling_rate,
+                    state_vector,
+                    state_history,
+                    observed_variables,
+                    interactive_objects,
+                )
+
+        elif len(sig.parameters) == 7:
+            new_controller_function = controller_function
+        else:
+            raise ValueError(
+                "The controller function must have 6 or 7 arguments. "
+                "The arguments must be in the following order: "
+                "(time, sampling_rate, state_vector, state_history, "
+                "observed_variables, interactive_objects, sensors)."
+                "Sensors argument is optional."
+            )
+        return new_controller_function
```

Comment:
It would change it for the airbrakes too, so its technically a breaking change. I do think having the sensors argument be optional can be good since not all rockets will have sensors. 

One thing I just thought of, maybe we should have all this arguments be passed as key word arguments in the user's `controller_function`. The user's controller functions would then have to receive `**kwargs` and just unpack it as needed:

```python
def controller_function(**kwargs):
    time = kwargs["time"]    
    sensors = kwargs["sensors"]

    ...
```

This would allow us to never need to care about how many params we are passing and in what order.

What do you think of this? It would be a breaking change, but only for the air brakes usage

Reply:
I think your proposal is really good.

Why would it be a breaking change? Couldn't you individually handle the case where the key "sensors" is not present in the kwargs dict?

Reply:
It would be a breaking change because any controller_function defined by a user would need to define its parameters differently